MonsterRace Step2

hsik0225 opened the pull request:
## Step1 피드백 사항

- 메소드의 적절한 위치가 어디인지 객체는 무엇인지에 대해 더 고민해 보세요
- input()도 별도 클래스에 존재하는 게 좋을 듯

![Package hsik0225](https://user-images.githubusercontent.com/56301069/74009768-d4e7fd80-497b-11ea-9cf5-fc82598eae5b.png)

## Step2 구현

- 클래스가 가지는 상태(State)와 행동(Behavior)에 맞춰 구현하려고 노력했습니다
- Input, Output을 별도의 클래스로 빼서 구현했습니다

hsik0225 committed:
Monster Race Step2

Monster Race Step1 클래스들을 쪼개보았다
기존 GameMachine, MonsterRace, Monster
변경 GameCencter, User, GameMachine, Input, Output, Monster

hsik0225 committed:
Monster Race Step2

간단한 스켈레톤 코드 작성

hsik0225 committed:
Monster Race Step2

Input 클래스구현
User 클래스 구현

hsik0225 committed:
Monster Race Step2

User 클래스가 필요하지 않은 것 같아 삭제

hsik0225 committed:
Monster Race Step2

GameCenter 클래스 삭제

hsik0225 committed:
MonsterRace Step2

게임 셋팅(몬스터의 수, 시도 횟수) 구현 완료
몬스터 생성 구현 완료

hsik0225 committed:
Moster Race Step2

Monster 클래스 Move 구현

hsik0225 committed:
Monster Race Step2

Monster 생성, Monster move 구현

hsik0225 committed:
Monster Race Step2

몬스터가 움직이는 거리를 잘못 생각했었다
몬스터 움직이는 거리 수정

hsik0225 committed:
Monster Race Step2

Step2 구현 완료
리팩토링 시작

hsik0225 committed:
Monster Race Step2

public에서 private으로 수정가능한 메소드들 수정

hsik0225 committed:
Monster Race Step2

컴파일 시 실행되지 않았다
생성하지 않았던 List<Monster> monsters 생성
정상 실행 확인

hsik0225 committed:
Monster Race Step2

몬스터 수에 0을 입력해도 실행이 되었다
정규표현식을 수정하여 1이상의 정수만 받도록 하였다

hsik0225 committed:
Monster Race Step2

UML 추가

ksundong commented on src/com/hsik0225/Input.java:
```diff
@@ -0,0 +1,38 @@
+
+public class Input {
+    private int MONSTER_COUNT;
+    private int CHALLENGE_COUNT;
```
naming convention에 어긋나는 것 같아요!

ksundong commented on src/com/hsik0225/Input.java:
```diff
@@ -0,0 +1,38 @@
+
+    private int inputDigit() {
+        String inputLine = scanner.nextLine();
+        while (!inputLine.matches("[1-9][0-9]*") || inputLine.equals("")) {
```
와우 정규식 😮

ksundong commented on src/com/hsik0225/MonsterRace.java:
```diff
@@ -1,19 +1,62 @@
-    public void run() {
-        Monster monster = new Monster();
+    private void makeMonsters(){
+        for (int index = 0; index < input.getMONSTER_COUNT(); index++) {
```
index일 필요가 있을까 싶어요.

ksundong reviewed (commented):
코드 잘 읽었습니다! 커밋 제목으로는 변경사항이 무엇인지 파악하기 힘드네요😅

hsik0225 committed:
Update Input.java

아무것도 입력하지 않았을 때도 정규표현식에 걸리게 됨
정규 표현식 수정

hsik0225 committed:
Depth 2단계에서 1단계로 수정

요구사항을 늦게봐서 다시 수정했습니다
setNumOfMove 함수에서 Stream을 안써도 되긴 하는데 함수 나누기가 애매해서 Stream을 사용했습니다.

honux77 commented:
커밋 메시지 다시 작성해서 PR 보내 주세요.
